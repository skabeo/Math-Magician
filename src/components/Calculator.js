import React from 'react';
import ButtonGrid from './ButtonGrid';

const Calculator = () => (
  <div className='calculatorCont white'>
    <h2 className='calculatorText'>Let's do some math!</h2>
    <ButtonGrid />
  </div>
);

export default Calculator;
