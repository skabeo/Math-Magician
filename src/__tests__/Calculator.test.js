import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../components/Calculator';

describe('Calculator component', () => {
  test('should render Calculator component correctly', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });

  test('Should render the value 1 when the 1 button is clicked', () => {
    const { getByText } = render(<Calculator />);
    const button = getByText('1');
    fireEvent.click(button);
    expect(getByText('1')).toBeInTheDocument();
  });
});
