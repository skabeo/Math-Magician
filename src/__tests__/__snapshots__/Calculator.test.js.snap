// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div>
  <div
    class="calculatorCont white"
  >
    <h2
      class="calculatorText"
    >
      Let's do some math!
    </h2>
    <div
      class="calculator-grid"
    >
      <input
        class="output"
        dir="rtl"
        placeholder="0"
      />
      <button
        type="button"
      >
        AC
      </button>
      <button
        type="button"
      >
        +/-
      </button>
      <button
        type="button"
      >
        %
      </button>
      <button
        class="orange"
        type="button"
      >
        รท
      </button>
      <button
        type="button"
      >
        7
      </button>
      <button
        type="button"
      >
        8
      </button>
      <button
        type="button"
      >
        9
      </button>
      <button
        class="orange"
        type="button"
      >
        x
      </button>
      <button
        type="button"
      >
        4
      </button>
      <button
        type="button"
      >
        5
      </button>
      <button
        type="button"
      >
        6
      </button>
      <button
        class="orange"
        type="button"
      >
        -
      </button>
      <button
        type="button"
      >
        1
      </button>
      <button
        type="button"
      >
        2
      </button>
      <button
        type="button"
      >
        3
      </button>
      <button
        class="orange"
        type="button"
      >
        +
      </button>
      <button
        class="two-spaces"
        type="button"
      >
        0
      </button>
      <button
        type="button"
      >
        .
      </button>
      <button
        class="orange"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator component should render Calculator component correctly 1`] = `
<div>
  <div
    class="calculatorCont white"
  >
    <h2
      class="calculatorText"
    >
      Let's do some math!
    </h2>
    <div
      class="calculator-grid"
    >
      <input
        class="output"
        dir="rtl"
        placeholder="0"
      />
      <button
        type="button"
      >
        AC
      </button>
      <button
        type="button"
      >
        +/-
      </button>
      <button
        type="button"
      >
        %
      </button>
      <button
        class="orange"
        type="button"
      >
        รท
      </button>
      <button
        type="button"
      >
        7
      </button>
      <button
        type="button"
      >
        8
      </button>
      <button
        type="button"
      >
        9
      </button>
      <button
        class="orange"
        type="button"
      >
        x
      </button>
      <button
        type="button"
      >
        4
      </button>
      <button
        type="button"
      >
        5
      </button>
      <button
        type="button"
      >
        6
      </button>
      <button
        class="orange"
        type="button"
      >
        -
      </button>
      <button
        type="button"
      >
        1
      </button>
      <button
        type="button"
      >
        2
      </button>
      <button
        type="button"
      >
        3
      </button>
      <button
        class="orange"
        type="button"
      >
        +
      </button>
      <button
        class="two-spaces"
        type="button"
      >
        0
      </button>
      <button
        type="button"
      >
        .
      </button>
      <button
        class="orange"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
